src/Controls/src/Core/VisualElement.cs

```diff
@@ -1199,16 +1193,19 @@ void IPropertyPropagationController.PropagatePropertyChanged(string propertyName
 			PropertyPropagationExtensions.PropagatePropertyChanged(propertyName, this, ((IVisualTreeElement)this).GetVisualChildren());
 		}
 
-		void SetSize(double width, double height)
+		void UpdateBoundsComponents(Rect bounds) 
 		{
-			if (Width == width && Height == height)
-				return;
+			BatchBegin();
```

Comment:
Should we move the code that sets the `_frame` local variable into this method?  I worry about developers messing up and calling this directly and then the `_frame` value won't match `x,y,width,height`. Unless the intent is for there to be a path where `_frame` doesn't match `X,Y,Width,Height` ?

Reply:
Done.